### ISSUE TITLE
Feature Request: Enable filament runout sensor when printing from Octoprint

### ISSUE THREAD

**Issue by evilC** [Snapmaker/SnapmakerController-IDEX#1]
# 🐞 feature request

### Is your feature request related to a problem? Please describe.
While printing from Octoprint, the filament runout sensor is not supported

### Describe the solution you'd like?
Enable filament runout sensor
I think this just requires `HOST_PROMPT_SUPPORT` to be enabled in `Configuration_adv.h` ?

**Comment by 747lulu747** [comment]
We are now doing compatibility with octoprint. It take some time. Please keep an eye out for commits on github

**Comment by TylonHH** [comment]
just uncommend in the firmware
`#define HOST_ACTION_COMMANDS`
see refernce

**Comment by Googliola** [comment]
> We are now doing compatibility with octoprint. It take some time. Please keep an eye out for commits on github

No commits since initial release. Issues get ignored for months. What's the plan?

**Comment by Googliola** [comment]
HOST_ACTION_COMMANDS enabled. Does not change a thing.

**Comment by TylonHH** [comment]
> HOST_ACTION_COMMANDS enabled. Does not change a thing.
How is it enabled? Do I need a special firmware? Because OctoPrint says still disabled with the latest official firmware

**Comment by Googliola** [comment]
No, I recompiled FW with HOST_ACTION_COMMANDS  enabled. Here's the binary (remove .txt):
[J1_V2.2.11_20230905.bin.txt](https://github.com/Snapmaker/SnapmakerController-IDEX/files/12558877/J1_V2.2.11_20230905.bin.txt)

**Comment by evilC** [comment]
I spoke to a Marlin dev about this, and enabling `HOST_ACTION_COMMANDS` will not fix it. Apparently SM rewrote part of the filament sensor code in Marlin or something. IIRC he said that because of this, some code will need to be changed in the UI, which we do not have access to, so we will be unable to fix it ourselves.

**Comment by TylonHH** [comment]
So that's why I also can’t use M2000 via USB.
